clar2wasm/src/wasm_generator.rs

```diff
@@ -1010,13 +1008,655 @@ impl WasmGenerator {
         size
     }
 
+    /// Serialize the value of type `ty` on the top of the data stack using
+    /// consensus serialization. Leaves the length of the data written on the
+    /// top of the data stack.
+    fn serialize_to_memory(
+        &mut self,
+        builder: &mut InstrSeqBuilder,
+        offset_local: LocalId,
+        offset: u32,
+        ty: &TypeSignature,
+    ) {
+        let memory = self
+            .module
+            .memories
+            .iter()
+            .next()
+            .expect("no memory found")
+            .id();
+
+        use clarity::vm::types::signatures::TypeSignature::*;
+        match ty {
```

Comment:
Would it be possible to extract functions from all the different branches, and call them from the match statement?
This is very long to read, and not super easy to navigate.